### ISSUE TITLE
`@typescript-eslint/object-curly-spacing`: Incorrectly warning for `{}`

### ISSUE THREAD

**Issue by k-funk** [eslint-stylistic/eslint-stylistic#1041]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

In the process of migrating from `@typescript-eslint/object-curly-spacing`

Seems like I'm getting a false positive for `React.PropsWithChildren`:

<img width="1026" height="98" alt="Image" src="https://github.com/user-attachments/assets/75f651d5-8b5d-457c-a44f-a16841a5b786" />

My config
```
'@stylistic/ts/object-curly-spacing': ['warn', 'always'],
```

I've also tried with `{ objectsInObjects: false/true }`

### Reproduction

Use React.PropsWithChildren with an empty object

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Can you provide a mini repo? Or at least share all the version information you're using. (Just run `npx envinfo --system --binaries --npmPackages '{eslint,@stylistic/*,@typescript-eslint/parser}'`)

**Comment by k-funk** [comment]
Tried to repro this in a minirepo, but was unable even with copy/pasted config.

here's my envinfo output:
```
  System:
    OS: macOS 26.1
    CPU: (16) arm64 Apple M4 Max
    Memory: 3.76 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 24.6.0 - /Users/kfunk/.nvm/versions/node/v24.6.0/bin/node
    npm: 11.5.1 - /Users/kfunk/.nvm/versions/node/v24.6.0/bin/npm
  npmPackages:
    @stylistic/eslint-plugin: ^5.5.0 => 5.5.0
    @typescript-eslint/parser: ^8.46.4 => 8.46.4
    eslint: ^9.39.1 => 9.39.1
```

Feel free to close this ticket until i'm able to repro for you in a repo.
